src/quacc/recipes/espresso/bands.py

```diff
@@ -0,0 +1,284 @@
+"""
+This module, 'bands.py', contains recipes for performing bands and fermi surface calculations using the
+bands.x and fs.x binaries from Quantum ESPRESSO via the quacc library.
+
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from ase.dft.kpoints import bandpath
+from pymatgen.io.ase import AseAtomsAdaptor
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+
+from quacc import job
+from quacc.calculators.espresso.espresso import EspressoTemplate
+from quacc.recipes.espresso._base import base_fn
+from quacc.utils.kpts import convert_pmg_kpts
+
+if TYPE_CHECKING:
+    from typing import Any, TypedDict
+
+    from ase.atoms import Atoms
+
+    from quacc.schemas._aliases.ase import RunSchema
+
+    class BandsSchema(TypedDict):
```

Comment:
Do `BandsSchema(TypedDict, total=False)` to indicate that not all three entries may necessarily be present.